From 1458863eb7975d297065fd974c397d361634f934 Mon Sep 17 00:00:00 2001
From: Nishant Sarmukadam <nishants@marvell.com>
Date: Thu, 30 Sep 2010 01:17:32 -0700
Subject: [PATCH 3/5] mwl8k: allow for padding in AP transmit descriptor

The AP transmit descriptor has padding of 20 bytes at the
end of the descriptor. This padding is not used by mwl8k driver
but it still has to be accounted for while calculating the size
of the descriptor in order to calculate start of next descriptor.
Allow for different transmit descriptor size due to padding for AP
transmit descriptor.

Signed-off-by: Nishant Sarmukadam <nishants@marvell.com>
---
 drivers/net/wireless/mwl8k.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/net/wireless/mwl8k.c b/drivers/net/wireless/mwl8k.c
index f152a25..de2dc40 100644
--- a/drivers/net/wireless/mwl8k.c
+++ b/drivers/net/wireless/mwl8k.c
@@ -1126,6 +1126,7 @@ struct mwl8k_tx_desc {
 	__le16 rate_info;
 	__u8 peer_id;
 	__u8 tx_frag_cnt;
+	__le32 pad[5];
 } __packed;
 
 #define MWL8K_TX_DESCS		128
-- 
1.7.0.1

